libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1064,19 +1068,23 @@ string YulUtilFunctions::clearStorageRangeFunction(Type const& _type)
 {
 	string functionName = "clear_storage_range_" + _type.identifier();
 
-	solAssert(_type.storageBytes() >= 32, "Expected smaller value for storage bytes");
-
 	return m_functionCollector.createFunction(functionName, [&]() {
 		return Whiskers(R"(
-			function <functionName>(start, end) {
+			function <functionName>(start, offset, end) {
+				if gt(offset, 0) {
+					let mask := <shr>(mul(8, sub(32, offset)), <ones>)
+					sstore(start, and(mask, sload(start)))
+				}
 				for {} lt(start, end) { start := add(start, <increment>) }
 				{
 					<setToZero>(start, 0)
 				}
 			}
 		)")
 		("functionName", functionName)
-		("setToZero", storageSetToZeroFunction(_type))
+		("ones", u256(-1).str())
```

Comment:
We tell users not to do that in Solidity and yet we do it ourselves ;)

Reply:
I'll fix this :)